apps/ui/src/queries/proposals.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  QueryClient,
+  useInfiniteQuery,
+  useQuery,
+  useQueryClient
+} from '@tanstack/vue-query';
+import { MaybeRef } from 'vue';
+import { getNames } from '@/helpers/stamp';
+import { getNetwork } from '@/networks';
+import { ProposalsFilter } from '@/networks/types';
+import { NetworkID, Proposal } from '@/types';
+
+type Filters = {
+  state?: MaybeRef<NonNullable<ProposalsFilter['state']>>;
+  labels?: MaybeRef<string[]>;
+};
+
+export const PROPOSALS_LIMIT = 20;
+export const PROPOSALS_SUMMARY_LIMIT = 6;
+
+export const PROPOSALS_KEYS = {
+  all: ['proposals'] as const,
+  space: (networkId: NetworkID, spaceId: string) =>
+    [...PROPOSALS_KEYS.all, `${networkId}:${spaceId}`] as const,
+  spaceSummary: (networkId: NetworkID, spaceId: string) =>
+    [...PROPOSALS_KEYS.space(networkId, spaceId), 'summary'] as const,
+  spaceList: (
+    networkId: NetworkID,
+    spaceId: string,
+    filters: Filters,
+    query?: MaybeRef<string>
+  ) =>
+    [
+      ...PROPOSALS_KEYS.space(networkId, spaceId),
+      'list',
+      { ...filters, query }
+    ] as const,
+  details: (networkId: NetworkID, spaceId: string) =>
+    [...PROPOSALS_KEYS.space(networkId, spaceId), 'detail'] as const,
+  detail: (
+    networkId: NetworkID,
+    spaceId: string,
+    proposalId: MaybeRef<string>
+  ) => [...PROPOSALS_KEYS.details(networkId, spaceId), proposalId] as const
+};
+
+async function withAuthorNames(proposals: Proposal[]) {
+  const names = await getNames(proposals.map(proposal => proposal.author.id));
+
+  return proposals.map(proposal => {
+    proposal.author.name = names[proposal.author.id];
+
+    return proposal;
+  });
+}
+
+function setProposalsDetails(
+  queryClient: QueryClient,
+  networkId: NetworkID,
+  spaceId: string,
+  proposals: Proposal[]
+) {
+  for (const proposal of proposals) {
+    queryClient.setQueryData(
+      PROPOSALS_KEYS.detail(
+        networkId,
+        spaceId,
+        proposal.proposal_id.toString()
+      ),
+      proposal
+    );
+  }
+}
+
+async function getProposals(
+  spaceId: string,
+  networkId: NetworkID,
+  { limit, skip }: { limit: number; skip: number },
+  filters?: ProposalsFilter,
+  query?: string
+) {
+  const metaStore = useMetaStore();
+  await metaStore.fetchBlock(networkId);
+
+  return withAuthorNames(
+    await getNetwork(networkId).api.loadProposals(
```

Comment:
Why no `setProposalsDetails` storing here ? Those proposals, even with filters, are still full valid proposal of the space

Reply:
This is function that is independent of vue-query that we use to fetch proposals. We use this function in queries (`useProposalsQuery` and `useProposalsSummaryQuery`) and call `setProposalsDetails` with its input. I tried to avoid leaking vue-query specific stuff outside of queries.